powerspikegg/go_gateway/gateway_test.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	fetcherpb "powerspike.gg/powerspikegg/rawdata/fetcher/service_gopb"
+	lolpb "powerspike.gg/powerspikegg/rawdata/public/leagueoflegends_gopb"
+	"testing"
+
+	"fmt"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+	"net"
+	"net/http"
+	"net/http/httptest"
+)
+
+type mockMatchFetcherServer struct {
+	fetcherpb.MatchFetcherServer
+
+	server *grpc.Server
+
+	// request contains the request sent by the client
+	matchRequests    []*fetcherpb.MatchRequest
+	summonerRequests []*lolpb.Summoner
+	// response contains the response to send back to the client
+	response *lolpb.MatchReference
+	// err contains an eventual error to throw back to the client
+	err error
+	// address contains the server address
+	address string
+}
+
+func newMockMatchFetcherServer() (*mockMatchFetcherServer, error) {
+	lis, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, fmt.Errorf("failed to listen: %v", err)
+	}
+	mock := &mockMatchFetcherServer{
+		address: lis.Addr().String(),
+	}
+	mock.server = grpc.NewServer()
+	fetcherpb.RegisterMatchFetcherServer(mock.server, mock)
+	// Register reflection service on gRPC server.
+	reflection.Register(mock.server)
+
+	go func(s *grpc.Server, l net.Listener) {
+		s.Serve(l)
+	}(mock.server, lis)
+	return mock, nil
+}
+
+// Mock of the match endpoint
+func (s *mockMatchFetcherServer) Match(_ context.Context, req *fetcherpb.MatchRequest) (*lolpb.MatchReference, error) {
+	s.matchRequests = append(s.matchRequests, req)
+
+	if s.err != nil {
+		return nil, s.err
+	}
+
+	return s.response, nil
+}
+
+// Resets the mock requests sent list
+func (s *mockMatchFetcherServer) reset() {
+	s.matchRequests = nil
+	s.summonerRequests = nil
+}
+
+// Mock of the match endpoint
+func (s *mockMatchFetcherServer) UpdateSummoner(req *lolpb.Summoner, resp fetcherpb.MatchFetcher_UpdateSummonerServer) error {
+	s.summonerRequests = append(s.summonerRequests, req)
+
+	if s.err != nil {
+		return s.err
+	}
+
+	resp.Send(s.response)
+
+	return nil
+}
+
+// TestSummonerCommand ensures summoner command correctly recovers summoners' matchs from the fetcher
+func TestGateway(t *testing.T) {
+	tt := []struct {
+		name             string
+		serverRequest    string
+		serverResponse   *lolpb.MatchReference
+		expectedResponse string
+		expectedStatus   int
+	}{
+		{
+			name:          "summoner query",
+			serverRequest: "/api/summoner/Rangork/EUW",
+			serverResponse: &lolpb.MatchReference{
+				// FIXME: weird to only have a single match possibility here, we should make two test structs,
```

Comment:
Remove this fixme and add a `TODO(archangel): add a test with multiple match references returned.` at the end of the function